### ISSUE TITLE
signal (6): Aborted -- memmory allocation bug.

### ISSUE THREAD

**Issue by dcarrera** [JuliaLang/julia#29775]
This looks like a memmory allocation bug in Julia. I have a package that failed automated tests:

https://s3.ap-south-1.amazonaws.com/juliarun-ci/94a04f005dd71ef91fcd54be6695832f035a298f/Planets_reverse_dependency_tests.html

I've tried to extract the more important messages. Long story short, it's a memmory corruption error that appears to come from somewhere deep inside Julila. There are too many files mentioned for me to figure out which one is really the culprit.

```
...
*** Error in `/opt/julia-1.0.0/bin/julia': malloc(): memory corruption: 0x0000000001b90470 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x777e5)[0x7f9cffe4a7e5]
/lib/x86_64-linux-gnu/libc.so.6(+0x8213e)[0x7f9cffe5513e]
/lib/x86_64-linux-gnu/libc.so.6(+0x82c0a)[0x7f9cffe55c0a]
/lib/x86_64-linux-gnu/libc.so.6(posix_memalign+0x11d)[0x7f9cffe5a71d]
/opt/julia-1.0.0/bin/../lib/libjulia.so.1(jl_gc_managed_malloc+0x4c)[0x7f9d0086203c]
/opt/julia-1.0.0/bin/../lib/libjulia.so.1(+0x6524d)[0x7f9d0082b24d]
...
======= Memory map: ========
00400000-00403000 r-xp 00000000 08:11 14027162                           /opt/julia-1.0.0/bin/julia
00602000-00603000 r--p 00002000 08:11 14027162                           /opt/julia-1.0.0/bin/julia
...
7f9ce4472000-7f9ce44a4000 r-xp 00000000 08:11 14944299                   /opt/julia-1.0.0/lib/julia/libspqr.so
7f9ce44a4000-7f9ce46a3000 ---p 00032000 08:11 14944299                   /opt/julia-1.0.0/lib/julia/libspqr.so
...
7f9ce46a5000-7f9ce4774000 r-xp 00000000 08:11 14944306                   /opt/julia-1.0.0/lib/julia/libumfpack.so
...
7f9ce4975000-7f9ce4976000 r-xp 00000000 08:11 14944304                   /opt/julia-1.0.0/lib/julia/libsuitesparse_wrapper.so
...
7f9ce4b77000-7f9ce4b79000 r-xp 00000000 08:11 14944305                   /opt/julia-1.0.0/lib/julia/libsuitesparseconfig.so
...
7f9ce4d7a000-7f9ce4d85000 r-xp 00000000 08:11 14944263                   /opt/julia-1.0.0/lib/julia/libccolamd.so
...
... <... many more *.so files ...>
...
signal (6): Aborted
in expression starting at /home/jrun/Planets/.julia/packages/Planets/Eah53/test/runtests.jl:2
gsignal at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
abort at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x7f9cffe4a7e9)
unknown function (ip: 0x7f9cffe5513d)
unknown function (ip: 0x7f9cffe55c09)
posix_memalign at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
jl_malloc_aligned at /buildworker/worker/package_linux64/build/src/julia_internal.h:829 [inlined]
jl_gc_managed_malloc at /buildworker/worker/package_linux64/build/src/gc.c:2838
_new_array_ at /buildworker/worker/package_linux64/build/src/array.c:108
jl_deserialize_value_array at /buildworker/worker/package_linux64/build/src/dump.c:1518
jl_deserialize_value at /buildworker/worker/package_linux64/build/src/dump.c:2023
jl_deserialize_value_array at /buildworker/worker/package_linux64/build/src/dump.c:1532
jl_deserialize_value at /buildworker/worker/package_linux64/build/src/dump.c:2023
jl_deserialize_struct at /buildworker/worker/package_linux64/build/src/dump.c:1859
jl_deserialize_value_any at /buildworker/worker/package_linux64/build/src/dump.c:1958 [inlined]
jl_deserialize_value at /buildworker/worker/package_linux64/build/src/dump.c:2155
jl_deserialize_value at /buildworker/worker/package_linux64/build/src/dump.c:1773
jl_deserialize_value_array at /buildworker/worker/package_linux64/build/src/dump.c:1532
jl_deserialize_value at /buildworker/worker/package_linux64/build/src/dump.c:2023
_jl_restore_incremental at /buildworker/worker/package_linux64/build/src/dump.c:3084
jl_restore_incremental at /buildworker/worker/package_linux64/build/src/dump.c:3141
_include_from_serialized at ./loading.jl:614
_require_from_serialized at ./loading.jl:678
_require at ./logging.jl:317
require at ./loading.jl:852
macro expansion at ./logging.jl:311 [inlined]
require at ./loading.jl:834
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2182
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1536 [inlined]
call_require at /buildworker/worker/package_linux64/build/src/toplevel.c:441 [inlined]
eval_import_path at /buildworker/worker/package_linux64/build/src/toplevel.c:476
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:633
jl_parse_eval_all at /buildworker/worker/package_linux64/build/src/ast.c:838
jl_load at /buildworker/worker/package_linux64/build/src/toplevel.c:821
include at ./boot.jl:317 [inlined]
include_relative at ./loading.jl:1038
include at ./sysimg.jl:29
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2182
include at ./client.jl:388
jl_fptr_trampoline at /buildworker/worker/package_linux64/build/src/gf.c:1829
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2182
do_call at /buildworker/worker/package_linux64/build/src/interpreter.c:324
eval_value at /buildworker/worker/package_linux64/build/src/interpreter.c:428
eval_stmt_value at /buildworker/worker/package_linux64/build/src/interpreter.c:363 [inlined]
eval_body at /buildworker/worker/package_linux64/build/src/interpreter.c:686
jl_interpret_toplevel_thunk_callback at /buildworker/worker/package_linux64/build/src/interpreter.c:799
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f9cef7d487f)
unknown function (ip: 0xffffffffffffffff)
jl_interpret_toplevel_thunk at /buildworker/worker/package_linux64/build/src/interpreter.c:808
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:787
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:734
jl_toplevel_eval_in at /buildworker/worker/package_linux64/build/src/builtins.c:622
eval at ./boot.jl:319
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2182
macro expansion at ./logging.jl:317 [inlined]
exec_options at ./client.jl:219
_start at ./client.jl:421
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2182
jl_apply at /buildworker/worker/package_linux64/build/ui/../src/julia.h:1536 [inlined]
true_main at /buildworker/worker/package_linux64/build/ui/repl.c:112
main at /buildworker/worker/package_linux64/build/ui/repl.c:233
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
_start at /opt/julia-1.0.0/bin/julia (unknown line)
Allocations: 13023623 (Pool: 13021646; Big: 1977); GC: 25
ERROR: Package Planets errored during testing
Stacktrace:
 [1] pkgerror(::String, ::Vararg{String,N} where N) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/Types.jl:120
 [2] macro expansion at ./logging.jl:313 [inlined]
 [3] #test#61(::Bool, ::Function, ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/Operations.jl:1283
 [4] #test at ./none:0 [inlined]
 [5] #test#44(::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:256
 [6] #test at ./none:0 [inlined]
 [7] #test#43 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:239 [inlined]
 [8] #test at ./none:0 [inlined]
 [9] #test#42(::Base.Iterators.Pairs{Symbol,Bool,Tuple{Symbol},NamedTuple{(:coverage,),Tuple{Bool}}}, ::Function, ::Array{String,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:238
 [10] #test at ./none:0 [inlined]
 [11] #test#41 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:237 [inlined]
 [12] (::getfield(Pkg.API, Symbol("#kw##test")))(::NamedTuple{(:coverage,),Tuple{Bool}}, ::typeof(Pkg.API.test), ::String) at ./none:0
 [13] top-level scope at none:0
```

**Comment by JeffBezanson** [comment]
Can this be triggered just by running the package tests for `Planets.jl`?

**Comment by dcarrera** [comment]
That seems to be the case. But it seems to depend on the local installation in ways that I haven't been able to figure out. See the very similar issue with #334. There the problem seemed to be with the CSV module and it seemed to be fixed by upgrading to 0.4.2 and then the tests passed in my local installation of Julia. So I changed `REQUIRE` to specify `CSV 0.4.2` and when I made a new release I got this other error from `JuliaCIBot`.
